text/0000-multi-part-examples-in-rustdoc/index.md

```diff
@@ -0,0 +1,231 @@
+- Feature Name: Multi-part examples in rustdoc
+- Start Date: 2021-02-09
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Support for multi-part examples in rustdoc, which consists of:
+  1. Running them as doc-tests
+  2. UI indicating members of multi-part examples in rendered docs
+
+# Motivation
+[motivation]: #motivation
+
+A long example often calls for lines of explanatory text interleaved within it.
+
+Currently, there are three means of achieving this:
+
+1. Use rust comments inside the long example.
+  Downsides:
+    1. No markdown
+    1. If a comment overflows the rendered code block horizontally, scrolling is possible, yet cumbersome, especially when the comment is far from the bottom, where the scroll bar happens to be.
+1. Split long examples into multiple code blocks and add the text between them.
+  Downsides:
+    1. Must disable doc-test-ing the example
+    1. Lack of UI indicating that the code blocks are multiple parts of a whole
+1. [The current documented workaround][current-documented-workaround]
+  Downsides:
+    1. Tedious. Macros may make this more reasonable, but still not nice.
+
+This RFC suggests a new feature in rustdoc that supports multi-part examples without any of the downsides listed above.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Multi-part examples
+
+A long example may benefit from being split up into multiple code blocks. The rendered documentation will indicate that each code block is a part of a whole. Also, doc-tests, run links, etc. will work as expected.
+
+The first block of a multi-part example must be tagged with `split-start`.
+Zero or more continuation blocks tagged with `split-continue` may follow.
+A final `split-end` block must be present.
+Between parts, any markdown may appear, including code blocks.
+However, no interleaving or nesting of multi-part examples is allowed.
+
+Not only _rust_ examples may be multi-part.
```

Comment:
How would you indicate that? `split-start,rust` or `rust,split-start` or something else?

Reply:
Examples are `rust` by default.

Reply:
I mean `rust` as a generic language tag — I could have been clearer by saying `split-start,javascript`. I think I feel a bit confused about why it says "Not only _rust_ examples may be multi-part."

Reply:
Does 592ede1eec0d1ef8bcd08e4dafb650a358f86ca0 resolve your concerns?